src/lib/hooks/useTokenList/utils.ts

```diff
@@ -1,36 +1,38 @@
 import { TokenInfo, TokenList } from '@uniswap/token-lists'
 import { WrappedTokenInfo } from 'state/lists/wrappedTokenInfo'
 
-type TokenMap = Readonly<{ [tokenAddress: string]: { token: WrappedTokenInfo; list?: TokenList } }>
-export type ChainTokenMap = Readonly<{ [chainId: number]: TokenMap }>
+type TokenInfoMap = Readonly<{ [tokenAddress in string]?: { token: WrappedTokenInfo; list?: TokenList } }>
+export type ChainTokenInfoMap = Readonly<{ [chainId in number]?: TokenInfoMap }>
 
 type Mutable<T> = {
   -readonly [P in keyof T]: Mutable<T[P]>
 }
 
-const mapCache = typeof WeakMap !== 'undefined' ? new WeakMap<TokenList | TokenInfo[], ChainTokenMap>() : null
+const mapCache = typeof WeakMap !== 'undefined' ? new WeakMap<TokenList | TokenInfo[], ChainTokenInfoMap>() : null
 
-export function tokensToChainTokenMap(tokens: TokenList | TokenInfo[]): ChainTokenMap {
+export function tokensToChainTokenInfoMap(tokens: TokenList | TokenInfo[]): ChainTokenInfoMap {
   const cached = mapCache?.get(tokens)
   if (cached) return cached
 
   const [list, infos] = Array.isArray(tokens) ? [undefined, tokens] : [tokens, tokens.tokens]
-  const map = infos.reduce<Mutable<ChainTokenMap>>((map, info) => {
+  const map = infos.reduce<Mutable<ChainTokenInfoMap>>((map, info) => {
     try {
       const token = new WrappedTokenInfo(info, list)
       if (map[token.chainId]?.[token.address] !== undefined) {
         console.warn(`Duplicate token skipped: ${token.address}`)
         return map
       }
-      if (!map[token.chainId]) {
-        map[token.chainId] = {}
+      const currentChainMap = map[token.chainId]
+      if (!currentChainMap) {
+        map[token.chainId] = { [token.address]: { token, list } }
+      } else {
+        currentChainMap[token.address] = { token, list }
       }
-      map[token.chainId][token.address] = { token, list }
       return map
     } catch {
       return map
     }
-  }, {}) as ChainTokenMap
+  }, {}) as ChainTokenInfoMap
```

Comment:
I think you could reduce the coercions by casting the initial object, and be slightly safer:

```suggestion
  }, {} as ChainTokenInfoMap)
```

Reply:
It seems to have been written this way to allow the ChainTokenInfoMap to be mutable while being constructed, and then casts it as the readonly version afterwards.